pkg/resources/oauth_integration_for_custom_clients_acceptance_test.go

```diff
@@ -0,0 +1,667 @@
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "blocked_roles_list.#", "3"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_issue_refresh_tokens", "true"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "oauth_refresh_token_validity", "86400"),
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "network_policy", sdk.NewAccountObjectIdentifier(networkPolicy.Name).Name()), // TODO(SNOW-999049): Fix during identifiers rework
+					// oauth_client_rsa_public_key and oauth_client_rsa_public_key_2 cannot be imported, because they're not available in SHOW and DESC outputs.
+					importchecks.TestCheckResourceAttrInstanceState(id.Name(), "comment", comment),
+				),
+			},
+			// change externally
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_OauthIntegrationForCustomClients/complete"),
+				ConfigVariables: m(true),
+				PreConfig: func() {
+					acc.TestClient().SecurityIntegration.UpdateOauthForClients(t, sdk.NewAlterOauthForCustomClientsSecurityIntegrationRequest(id).WithUnset(
+						*sdk.NewOauthForCustomClientsIntegrationUnsetRequest().
+							WithEnabled(true).
+							WithNetworkPolicy(true).
+							WithOauthUseSecondaryRoles(true).
+							WithOauthClientRsaPublicKey(true).
+							WithOauthClientRsaPublicKey2(true),
+					))
+				},
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.PrintPlanDetails("snowflake_saml2_integration.test", "enabled", "network_policy", "oauth_use_secondary_roles", "oauth_client_rsa_public_key", "oauth_client_rsa_public_key_2"),
+
+						// Not showing any drift and change, but alter is run underneath to set the values back
+						// planchecks.ExpectDrift("snowflake_saml2_integration.test", "enabled", sdk.String("true"), sdk.String("false")),
+						// planchecks.ExpectDrift("snowflake_saml2_integration.test", "network_policy", sdk.String(sdk.NewAccountObjectIdentifier(networkPolicy.Name).Name()), sdk.String("")),
+						// planchecks.ExpectDrift("snowflake_saml2_integration.test", "oauth_use_secondary_roles", sdk.String("true"), sdk.String("false")),
+						// planchecks.ExpectDrift("snowflake_saml2_integration.test", "oauth_client_rsa_public_key", sdk.String(key), sdk.String("")),
+						// planchecks.ExpectDrift("snowflake_saml2_integration.test", "oauth_client_rsa_public_key_2", sdk.String(key), sdk.String("")),
+
+						// planchecks.ExpectChange("snowflake_saml2_integration.test", "enabled", tfjson.ActionUpdate, sdk.String("false"), sdk.String("true")),
+						// planchecks.ExpectChange("snowflake_saml2_integration.test", "network_policy", tfjson.ActionUpdate, sdk.String(""), sdk.String(sdk.NewAccountObjectIdentifier(networkPolicy.Name).Name())),
+						// planchecks.ExpectChange("snowflake_saml2_integration.test", "oauth_use_secondary_roles", tfjson.ActionUpdate, sdk.String("false"), sdk.String("true")),
```

Comment:
Previously it had "NONE" value in state, what's the value after UNSET: empty or "NONE"?

Reply:
Yeah, should've set it to something else because that's the default value for this field